lightning/pkg/importer/precheck_impl.go

```diff
@@ -1433,3 +1435,73 @@ func hasDefault(col *model.ColumnInfo) bool {
 	return col.DefaultIsExpr || col.DefaultValue != nil || !mysql.HasNotNullFlag(col.GetFlag()) ||
 		col.IsGenerated() || mysql.HasAutoIncrementFlag(col.GetFlag())
 }
+
+type pdTiDBFromSameClusterCheckItem struct {
+	db            *sql.DB
+	pdAddrsGetter func(context.Context) []string
```

Comment:
I recommend replacing these fields in the struct like:
```go
getPDAddrsFromSQL func(context.Context) []string
getPDAddrsFromPD func(context.Context) []string
```
This change improves flexibility and provides a clearer abstraction by distinguishing between upstream and downstream PD address getters. Additionally, it simplifies unit testing, as you can now mock and test the two getters independently.

Reply:
1.

> distinguishing between upstream and downstream PD address

There's no upstream PD addresses. However I understand your meaning is check the two ways to get PD addresses.

2. I don'n know how CDC can reuse this struct `pdTiDBFromSameClusterCheckItem`, because the intersection assertion can not be configurated, and CDC wants to assert non-intersection.